include/extractor/guidance/toolkit.hpp

```diff
@@ -285,6 +288,60 @@ inline std::string applyAccessTokens(std::string lane_string, const std::string
     return result_string;
 }
 
+LaneID inline numLanesToTheRight(const engine::guidance::RouteStep &step)
+{
+    return step.intersections.front().lanes.first_lane_from_the_right;
+}
+
+LaneID inline numLanesToTheLeft(const engine::guidance::RouteStep &step)
+{
+    LaneID total = step.intersections.front().lane_description.size();
+    return total - (step.intersections.front().lanes.lanes_in_turn +
+                    step.intersections.front().lanes.first_lane_from_the_right);
+}
+
+template <class Iterator>
+inline boost::iterator_range<Iterator>
+lanesToTheLeft(const util::guidance::LaneTuple &lanes,
+               const extractor::guidance::TurnLaneDescription &lane_description)
+{
+    const auto farthest_left_turn_lane =
+        lane_description.size() - (lanes.first_lane_from_the_right + lanes.lanes_in_turn);
+    // check that lanes even exist left of the turn lanes
+    if (farthest_left_turn_lane != 0)
+    {
+        // return a range from the farthest left lane, to the first lane left of turn lanes
+        const auto first_left_of_turn_lanes = lane_description.begin() + (farthest_left_turn_lane);
+        auto out_range =
+            boost::make_iterator_range(lane_description.begin(), first_left_of_turn_lanes);
+        return out_range;
+    }
+    else
+    {
+        return boost::make_iterator_range(lane_description.end(), lane_description.end());
+    }
+}
```

Comment:
Conceptually, can't you use your functions above to get the number of lanes left / right of the turn and then do

    numLeft = numLanesToLeft(step);
    return make_it_range(begin(lanes), begin(lanes) + numLeft);

if numLeft is zero you get two iterators back where first == last, so an empty range.
If not you slice the lanes into a sub-range.